app/src/main/java/org/thoughtcrime/securesms/ui/Components.kt

```diff
@@ -511,4 +516,55 @@ fun LoadingArcOr(loading: Boolean, content: @Composable () -> Unit) {
     AnimatedVisibility(!loading) {
         content()
     }
+}
+
+
+@Composable
+fun SearchBar(
+    query: String,
+    onValueChanged: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    placeholder: String? = null,
+    enabled: Boolean = true,
+    backgroundColor: Color = LocalColors.current.background
+) {
+    BasicTextField(
+        singleLine = true,
+        value = query,
+        onValueChange = onValueChanged,
+        enabled = enabled,
+        decorationBox = { innerTextField ->
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(backgroundColor, RoundedCornerShape(100))
+            ) {
+                Image(
+                    painterResource(id = R.drawable.ic_search_24),
+                    contentDescription = null,
+                    colorFilter = ColorFilter.tint(
+                        LocalColors.current.textSecondary
+                    ),
+                    modifier = Modifier
+                        .padding(horizontal = 16.dp, vertical = 8.dp)
+                        .size(24.dp)
```

Comment:
Can use the LocalDimensions values for the padding and the newly created (from previous comments) value for icon sizes for the 24.dp